crates/typst/src/diag.rs

```diff
@@ -298,27 +310,51 @@ pub type HintedStrResult<T> = Result<T, HintedString>;
 /// A string message with hints.
 #[derive(Debug, Clone, Eq, PartialEq, Hash)]
 pub struct HintedString {
+    /// The string's components as a single vector of strings.
+    /// The first element of the vector contains the message.
+    /// The remaining elements are the hints.
+    /// This is done to reduce the size of a HintedString.
+    /// The vector is guaranteed to not be empty.
+    components: EcoVec<EcoString>,
```

Comment:
Since it's just a single field, we can use `pub struct HintedString(EcoVec<EcoString>);`